tests/testthat/test-unnest-longer.R

```diff
@@ -173,22 +256,39 @@ test_that("can unnest multiple columns (#740)", {
 
 test_that("tidyverse recycling rules are applied when unnesting multiple cols", {
   df <- tibble(a = list(1L, 3:4), b = list(1:2, 4L))
+
   out <- unnest_longer(df, c(a, b))
   expect_identical(out$a, c(1L, 1L, 3L, 4L))
   expect_identical(out$b, c(1L, 2L, 4L, 4L))
 })
 
+test_that("tidyverse recycling rules are applied after `keep_empty`", {
+  df <- tibble(a = list(NULL, 3:4), b = list(1:2, 4L))
+
+  expect_snapshot(error = TRUE, {
+    unnest_longer(df, c(a, b))
+  })
+
+  out <- unnest_longer(df, c(a, b), keep_empty = TRUE, indices_include = TRUE)
+
+  expect_identical(out$a, c(NA, NA, 3L, 4L))
+  expect_identical(out$a_id, c(1L, 1L, 1L, 2L))
```

Comment:
These values confused me for a while because I though they must be out indexes; but they're inner indexes, some of which point to elements that don't exist. (`NULL[[1]]` is ok by coincidence, but `double()[[1]]` is not).

Reply:
These are now `NA` instead of `1L`